lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -642,4 +572,129 @@ defmodule Trento.Domain.SapSystem do
         false
     end)
   end
+
+  defp abap_or_messageserver?(features) do
+    String.contains?(features, ["MESSAGESERVER", "ABAP"])
+  end
+
+  defp abap_and_messageserver?(features) do
+    String.contains?(features, "ABAP") and String.contains?(features, "MESSAGESERVER")
+  end
+
+  defp maybe_emit_sap_system_registered_event(
```

Comment:
Let's respect the order of the functions by precedence.
So:
1. `emit_application_instance_registered_or_application_instance_health_changed`
2. `maybe_emit_sap_system_registered_event`
etc

Reply:
done